src/core/Akka.Cluster/Cluster.cs

```diff
@@ -561,13 +565,65 @@ internal IActorRef ClusterCore
             }
         }
 
+        /// <summary>
+        /// INTERNAL API.
+        ///
+        /// Used for logging important messages with the cluster's address appended.
+        /// </summary>
+        internal class InfoLogger
```

Comment:
Created an internal abstraction for the `InfoLogger` in order to try to follow roughly what the JVM does, but also because I imagined some use cases where we might want to pass it around in the future.